src/scripting/lua_widget_attributes.cpp

```diff
@@ -200,16 +227,37 @@ WIDGET_SETTER("value_compat,selected_index", int, gui2::listbox)
 	w.select_row(value - 1);
 }
 
+WIDGET_GETTER("value_compat,value", std::string, gui2::menu_button)
+{
+    return w.get_value_string();
+}
+
+WIDGET_SETTER("value_compat,value", std::string, gui2::menu_button)
+{
+	luaL_error(L, "attempt to modify read-only attribute 'value'");
+}
+
+WIDGET_SETTER("value_compat,selected_index", int, gui2::menu_button)
```

Comment:
`value_compat` is for backwards compatibility with the older API. Do we really need it for new widgets that were never supported in the old API?

Reply:
Not a clue here.  One of those things I cut and paste.  Tried to figure out what it was once.

Reply:
Removing it means that `wesnoth.set_dialog_value(3, "path", "to", "menu_button")` will not work. But that never worked in the first place, and that API is deprecated, so who cares, right?